lib/ansible/plugins/inventory/__init__.py

```diff
@@ -434,23 +436,21 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False, fetch_h
                             continue
 
                         new_raw_group_names = []
-                        if isinstance(key, string_types):
-                            # if key is empty, 'default_value' will be used as group name
-                            if key == '' and default_value_name is not None:
-                                new_raw_group_names.append(default_value_name)
-                            else:
-                                new_raw_group_names.append(key)
+                        if use_default:
+                            new_raw_group_names.append(default_value_name)
+                        elif isinstance(key, string_types):
+                            new_raw_group_names.append(key)
                         elif isinstance(key, list):
                             for name in key:
                                 # if list item is empty, 'default_value' will be used as group name
-                                if name == '' and default_value_name is not None:
+                                if (name is None or name == '') and default_value_name is not None:
```

Comment:
change to `name in DEFAULTED`  aka `(None, '')` and we can make the several checks the same? if we ever expand we can just expand the variable.

Reply:
`key in DEFAULTED` above and `gval in DEFAULTED` below

Reply:
Sorry I missed this comment, have added a tuple to check with `val in ...`.